content/en/blog/2025/how-to-name-your-span-attributes.md

```diff
@@ -0,0 +1,280 @@
+---
+title: How to Name Your Span Attributes
+linkTitle: How to Name Your Span Attributes
+date: 2025-08-27
+author: >-
+  [Juraci Paixão Kröhling](https://github.com/jpkrohling) (OllyGarden)
+canonical_url: https://blog.olly.garden/how-to-name-your-span-attributes
+cSpell:ignore: interoperability jpkrohling OllyGarden shopify
+---
+
+Welcome to the second installment in our series on OpenTelemetry naming best
+practices. In our [previous post](/blog/2025/how-to-name-your-spans/), we
+explored how to name spans using the `{verb} {object}` pattern. Today, we're
+diving into span attributes—the rich contextual data that transforms your traces
+from simple operation logs into powerful debugging and analysis tools.
+
+This guide targets developers who are:
+
+- **Instrumenting their own applications** with custom spans and attributes
+- **Enriching telemetry** beyond what auto-instrumentation provides
+- **Creating libraries** that others will instrument
+
+The attribute naming decisions you make directly impact the usability and
+maintainability of your observability data. Let's get them right.
+
+## Start with Semantic Conventions
+
+Here's the most important rule that will save you time and improve
+interoperability: **if an OpenTelemetry semantic convention exists and the
+semantics match your use case, use it**.
```

Comment:
Maybe this is a good spot to point people to the [semconv attribute registry](https://opentelemetry.io/docs/specs/semconv/registry/attributes/) if they need to search for a semantic convention to match their use case?